import-scripts/clear-persistence-caches-for-changed-studies.sh

```diff
@@ -0,0 +1,450 @@
+    rm -f $tmp_study_subdir/*.$STUDY_CACHE_INVALIDATION_NEEDED_SUFFIX $tmp_study_subdir/*.$STUDY_CACHE_INVALIDATION_SUCCEEDED_SUFFIX
+    touch "$cache_reset_needed_filepath"
+}
+
+# returns 0 if full cache reset trigger is present
+# usage : database_is_triggered_as_needing_reset db_label
+function database_is_triggered_as_needing_reset() {
+    db="$1"
+    tmp_study_subdir="$TMP_DIR_ROOT/$db"
+    cache_reset_needed_filepath="$tmp_study_subdir/$CACHE_RESET_NEEDED_FILENAME"
+    if [ -f "$cache_reset_needed_filepath" ] ; then
+        return 0
+    fi
+    return 1
+}
+
+# examines age of each trigger file in specified db tmp_subdir and returns 0 if any is older than MAXIMUM_TIME_FOR_STUDY_RESET_BEFORE_ESCALATION
+# usage : any_trigger_has_exceeded_maximum_wait db_label
+function any_trigger_has_exceeded_maximum_wait() {
+    db="$1"
+    tmp_study_subdir="$TMP_DIR_ROOT/$db"
+    # get current epoc seconds
+    tmp_modified_filepath=$( mktemp --tmpdir=$TMP_DIR_ROOT tmp_timestamp_filepath.tmpXXXXXXXX)
+    now_seconds_since_epoc=$( stat -c%Y "$tmp_modified_filepath" )
+    rm -f "$tmp_modified_filepath"
+    # get oldest trigger file epoc seconds
+    for triggerfilepath in $tmp_study_subdir/*.$STUDY_CACHE_INVALIDATION_NEEDED_SUFFIX ; do
+        if [ -f "$triggerfilepath" ] ; then
+            trigger_seconds_since_epoc=$( stat -c%Y "$triggerfilepath" )
+            age_in_seconds=$(( $now_seconds_since_epoc - $trigger_seconds_since_epoc ))
+            if [ $age_in_seconds -gt $MAXIMUM_TIME_FOR_STUDY_RESET_BEFORE_ESCALATION ] ; then
+                return 0
+            fi
+        fi
+    done
+    return 1
+}
+
+# tests whether there is a defined cache api key and cache api uri for the database
+# usage : database_can_clear_per_study_cache db_label
+function database_can_clear_per_study_cache() {
+    db="$1"
+    if [ -z ${portal_db_to_cache_api_key[$db]} ] ; then
+        # cannot clear study without cache api key
+        return 1
+    fi
+    if [ -z ${PORTAL_DB_TO_PORTAL_API_BASE_URI[$db]} ] ; then
+        # cannot clear study without api path
+        return 1
+    fi
+    return 0
+}
+
+# gets the latest study info for a specified db and sees if anything has changed since the last execution of this script (trigger files created as side effect)
+# usage : search_for_studies_with_new_modifications_in_db db_label
+function search_for_studies_with_new_modifications_in_db() {
+    db="$1"
+    tmp_study_subdir="$TMP_DIR_ROOT/$db"
+    create_directory_if_necessary "$tmp_study_subdir"
+    previous_study_info_from_db_filepath="$TMP_DIR_ROOT/${db}_study_info"
+    new_study_info_from_db_filepath="$TMP_DIR_ROOT/${db}_study_info_new"
+    if ! [ -f "$previous_study_info_from_db_filepath" ] ; then
+        # if we have lost the previous study info, we cannot tell what has been modified. Prime the study info and trigger a full cache reset
+        write_current_study_list_from_db "$db" "$previous_study_info_from_db_filepath"
```

Comment:
Check return status here.

Reply:
Done. If the mysql command fails, a log message is generated and this function sees the failure status and returns failure too.